index.js

```diff
@@ -0,0 +1,167 @@
+var instance_skel  = require('../../instance_skel');
+const WebSocket    = require('ws');
+var objectPath     = require('object-path');
+
+class instance extends instance_skel {
+	constructor(system, id, config) {
+		super(system, id, config);
+
+		this.ws = null;
+		this.subscriptions = new Map();
+
+		this.actions();
+		this.initFeedbacks();
+		this.subscribeFeedbacks()
+
+		if (!this.config) {
+			return this;
+		}
+
+		this.initWebSocket();
+
+		return this;
+	}
+
+	updateConfig(config) {
+		this.config = config;
+		this.setVariableDefinitions([{}]);
+		this.initWebSocket();
+	}
+
+	initWebSocket() {
+		var ip = this.config.host;
+		var port = this.config.port;
+		this.status(this.STATUS_UNKNOWN)
+		if (!ip || !port) {
+			this.status(this.STATUS_ERROR)
+			return;
+		}
+		this.ws = new WebSocket(`ws://${ip}:${port}`);
+
+		this.ws.on('open', () => {
+			this.log('debug', `Connection opened`);
+			this.status(this.STATUS_OK)
+		});
+		this.ws.on('close', () => {
+			this.log('debug', `Connection closed`);
+			this.status(this.STATUS_ERROR)
+		});
+
+		this.ws.on('message', (data) => {
+			this.messageReceivedFromWebSocket(data);
+		});
+
+		this.ws.on('error', (data) => {
+			this.log('error', `WebSocket error: ${data}`);
+		});
+	}
+
+	messageReceivedFromWebSocket(data) {
+		this.log('debug', `Message received: ${data}`);
+		var msgValue = null;
+		try {
+			msgValue = JSON.parse(data);
+		} catch (e) {
+			msgValue = data;
+		}
+		this.subscriptions.forEach((subscription) => {
+			if (subscription.variableName === '') {
+				return;
+			}
+			if (subscription.subpath === '') {
+				this.setVariable(subscription.variableName, typeof msgValue === 'object' ? JSON.stringify(msgValue) : msgValue);
+			} else if (typeof msgValue === 'object' && objectPath.has(msgValue, subscription.subpath)) {
+				let value = objectPath.get(msgValue, subscription.subpath);
+				this.setVariable(subscription.variableName, typeof value === 'object' ? JSON.stringify(value) : value);
+			}
+		});
+	}
+
+	action(action) {
+		if (action.action === 'send_command') {
+			this.log('debug', `Message sent: ${action.options.data}`);
+			this.ws.send(action.options.data+"\r\n");
```

Comment:
This can be done as a `callback` to the action, similarly to as shown in your feedback definition, if you'd like